src/practitest_firecracker/query_dsl.cljc

```diff
@@ -0,0 +1,140 @@
+                           (return-error "Syntax error: 'join' must have one argument" query))
+    'split               (if (= 2 (count args))
+                           (let [quoted    #?(:clj  (string/escape (first args) char-escape-string)
+                                              :cljs (first args))
+                                 complied  #?(:clj  (java.util.regex.Pattern/compile quoted)
+                                              :cljs (js/RegExp. quoted))]
+                             (string/split (second args) complied))
+                           (return-error "Syntax error: 'split' must have two arguments" query))
+    'get               (if (= 2 (count args))
+                         (first (take 1 (drop (- (parse-int (first args)) 1) (last args))))
+                         (return-error "Syntax error: 'get' must have two arguments" query))
+    'trim               (if (= 1 (count args))
+                          (if (string? (first args))
+                            (string/trim (first args))
+                            (return-error "Syntax error: 'trim' should be applied on a string" query))
+                          (return-error "Syntax error: 'trim' must have only one argument" query))
+    (return-error #?(:clj (format "Syntax error: unsupported function '%s'" op)
+                     :cljs (str "Syntax error: unsupported function: " op)) query)))
+
+(defn eval-query [entity-hash query]
+  (if (map? query)
+    (let [{:keys [op args]} query
+          args              (map (partial eval-query entity-hash) args)]
+      (parse-methods op args query))
+    #?(:cljs  (let [key (keyword (string/join (drop 1 (str query))))]
+                   (cond
+                     (= '?field query)                     entity-hash
+                     (and (not (= entity-hash nil))
+                          (contains? entity-hash key))     (key entity-hash)
+                     (number? query)                       query
+                     :else                                 (str query)))
+       :clj (let [key (keyword (string/join (drop 1 (str query))))]
+              (cond
+                (or (= :test-suite-name key)
+                    (= :test-case-name key))          (:name entity-hash)
+                (and (not (= entity-hash nil))
+                     (contains? entity-hash key))     (key entity-hash)
+                (string/starts-with? (str query) "?") (str "")
+                :else                                 (str query))))))
+
+(defn read-query [s]
+  (let [query    (edn/read-string s)
+        compiler (fn compile-query [query]
+                   (if (list? query)
+                     (let [[op & args] query]
+                       {:op   (compile-query op)
+                        :args (vec (map compile-query args))})
+                     query))]
+    #?(:cljs (if (and (not (number? query)) (not (string? query)))
```

Comment:
Am I missing something here ? Those two branches look the same to me. Only thing that's different is `type` function on CLJ side, is there some hidden meta type information used ?

```
(or (= java.lang.Long (type query))
      (= java.lang.Double (type query))
      (and (= java.lang.String (type query)))
```

should be

`(or (string? query) (number? query))`

But maybe I'm missing something here.

Reply:
you are correct, I will merge this code into one line and will use "de morgan" on it